toolchain/check/handle_name.cpp

```diff
@@ -72,6 +72,26 @@ static auto GetClassElementIndex(Context& context, SemIR::InstId element_id)
                  << " in class element name";
 }
 
+// Returns whether `function_id` is an instance method, that is, whether it has
+// an implicit `self` parameter.
+static auto IsInstanceMethod(const SemIR::File& sem_ir,
+                             SemIR::FunctionId function_id) -> bool {
+  auto& function = sem_ir.functions().Get(function_id);
+  for (auto param_id :
+       sem_ir.inst_blocks().Get(function.implicit_param_refs_id)) {
+    auto inst = sem_ir.insts().Get(param_id);
+    if (auto addr = inst.TryAs<SemIR::AddrPattern>()) {
+      inst = sem_ir.insts().Get(addr->inner_id);
+    }
+    if (auto param = inst.TryAs<SemIR::Param>();
+        param && param->name_id == SemIR::NameId::SelfValue) {
```

Comment:
This code matches what's in convert.cpp, differing only on what's done with `param`. Not sure it's worth a like `TryAsSelfValueParam` or similar, but might be worth considering.

Reply:
Maybe we can take another look once `ConvertCallArgs` is also handling deduced parameters. They might look less similar at that point.